### PR TITLE
Test images with different utilization

### PR DESCRIPTION
**Pull request by nirs**
Enhance createTestImage to create image with configurable utilization and add 2 additional test:

- 0% utilization - this the best case when all work can be eliminated without reading any cluster from storage.

- 100% utilization - this is the worst case for both compressed and uncompressed images, when we have to read and decompress all clusters.

The test reveals that we have 2 issues:

- Reading empty uncompressed image is almost 3 times slower compare to reading fully allocated image (3430 MB/s vs 8783 MB/s). Even with the inefficient io.Reader interface, we except to perform zero read many times faster compared with reading actual cluster from storage.

- Decompression is extremely slow (39.28 MB/s for fully allocated image).

Example run:

    % go test -bench Read -count=1
    BenchmarkRead0p/qcow2-12         	      14	  78238414 ns/op	3430.99 MB/s	 1051160 B/op	      39 allocs/op
    BenchmarkRead0p/qcow2_zlib-12    	      14	  78577923 ns/op	3416.17 MB/s	 1051733 B/op	      39 allocs/op
    BenchmarkRead50p/qcow2-12        	      21	  54889353 ns/op	4890.48 MB/s	 1183231 B/op	      45 allocs/op
    BenchmarkRead50p/qcow2_zlib-12   	       1	3466799292 ns/op	  77.43 MB/s	736076536 B/op	  178764 allocs/op
    BenchmarkRead100p/qcow2-12       	      38	  30562127 ns/op	8783.27 MB/s	 1182901 B/op	      45 allocs/op
    BenchmarkRead100p/qcow2_zlib-12  	       1	6834526167 ns/op	  39.28 MB/s	1471530256 B/op	  357570 allocs/op

### CODE REVIEW

File: qcow2reader_test.go [hunk 1]
```diff
@@ -22,10 +22,73 @@ const (
 	CompressionTypeNone = qcow2.CompressionType(255)
 )
 
-func BenchmarkRead(b *testing.B) {
+// Benchmark sparse image with 10% utilization.  This is the best case when we
+// don't have to read any cluster from storage.
+func BenchmarkRead0p(b *testing.B) {
```
Comment (AkihiroSuda) [inline qcow2reader_test.go:27]:
```suggestion
func BenchmarkRead10p(b *testing.B) {
```

Comment (nirs) [reply qcow2reader_test.go:27]: This is an error in the code - leftover from previous version. I wanted to test 0% utilization.

Comment (AkihiroSuda) [reply qcow2reader_test.go:27]: The comment line should be updated?

Comment (nirs) [reply qcow2reader_test.go:27]: The comment is fixed now.

**Review by AkihiroSuda — APPROVED** [review]
Thanks

**Pull request merged**